taxii2client/common.py

```diff
@@ -130,18 +130,37 @@ def _to_json(resp):
         ), e)
 
 
+def _grab_total_items_from_resource(resp):
+    """Returns number of objects in bundle/envelope"""
+    resp = _to_json(resp)
+    return len(resp.get("objects", []))
+
+
 def _grab_total_items(resp):
-    """Extracts the Total elements available on the Endpoint making the request"""
+    """Extracts the total elements (from HTTP Header) available on the Endpoint making the request"""
     try:
         results = re.match(r"^items (\d+)-(\d+)/(\d+)$", resp.headers["Content-Range"])
-        return int(results.group(2)) - int(results.group(1)) + 1, int(results.group(3))
+        if results:
+            return int(results.group(2)) - int(results.group(1)) + 1, int(results.group(3))
+
+        results = re.match(r"^items (\d+)-(\d+)/\*$", resp.headers["Content-Range"])
+        if results:
+            return int(results.group(2)) - int(results.group(1)) + 1, float("inf")
+
+        results = re.match(r"^items \*/\*$", resp.headers["Content-Range"])
+        if results:
+            return float("inf"), float("inf")
+
+        results = re.match(r"^items \*/(\d+)$", resp.headers["Content-Range"])
```

Comment:
We are assuming the "unit" of the range header is `items` but does the TAXII spec require this? Should we accept other units?

I also noticed we don't have any tests for `_grab_total_items()` or `as_pages()`.

Reply:
Yes, TAXII 2.0 defines `items` as the range unit in section 3.4 and its the mandatory to implement unit. We shouldn't accept any other unit.

Reply:
We don't have any tests yet, I wanted someone to comment on the new behavior for the pagination requests (aside from more support) for both specs.

Reply:
added tests for both v20 and v21 `as_pages()`